src/components/EventDatesComponent/index.js

```diff
@@ -0,0 +1,215 @@
+    );
+
+    let currentContainer, currentCardsContainer;
+
+    const component = document.createElement("div");
+    component.classList.add("event-dates_component");
+
+    headerElements.forEach((header) => {
+      if (header.tagName === "H1") {
+        currentContainer = this._createElement("div", "event-dates_container");
+        component.appendChild(currentContainer);
+
+        const headerText = header.textContent.trim();
+        const headerDiv = this._createElement("div", "event-dates_header");
+        currentContainer.appendChild(headerDiv);
+
+        if (headerText.toLowerCase().startsWith("[bonus]")) {
+          const bonusSpan = this._createElement(
+            "span",
+            "event-dates_bonus-span"
+          );
+          bonusSpan.textContent = "Bonus";
+          headerDiv.appendChild(bonusSpan);
+
+          let next = header.nextElementSibling;
+          while (next && !/^H1$/.test(next.tagName)) {
+            currentContainer.appendChild(next);
+            next = header.nextElementSibling;
+          }
+        }
+
+        headerDiv.appendChild(document.createTextNode(headerText));
+
+        const cardsWrapper = this._createElement(
+          "div",
+          "event-date-cards_wrapper"
+        );
+        currentContainer.appendChild(cardsWrapper);
+
+        currentCardsContainer = this._createElement(
+          "div",
+          "event-date-cards_container"
+        );
+        cardsWrapper.appendChild(currentCardsContainer);
+      } else {
+        const card = this._createElement("div", "event-date-card");
+        currentCardsContainer.appendChild(card);
+
+        const titleDiv = this._createElement("div", "event-date-card_title");
+        titleDiv.textContent = header.textContent.trim();
+        card.appendChild(titleDiv);
+
+        let nextElement = header.nextElementSibling;
+        while (nextElement && !nextElement.tagName.startsWith("H")) {
+          if (nextElement.tagName === "UL") {
+            [...nextElement.querySelectorAll("li")]
+              .map((list, index) =>
+                this._convertListItemToCardInfo(list, index)
+              )
+              .forEach((cardInfo) => card.appendChild(cardInfo));
+          }
+
+          nextElement = nextElement.nextElementSibling;
+        }
+      }
+    });
+
+    return component;
+  }
+
+  _convertListItemToCardInfo(item, index) {
+    const cardInfo = this._createElement("div", "event-date-card_info");
+
+    if (index === 0) {
+      cardInfo.appendChild(this._createCalendarIcon());
+      cardInfo.style.fontWeight = "bold";
+    } else if (index === 1) {
+      cardInfo.appendChild(this._createClockIcon());
+    }
+
+    const itemText = item.textContent.trim();
+    cardInfo.appendChild(document.createTextNode(itemText));
+
+    return cardInfo;
+  }
+
+  _createElement(tag, className) {
+    const element = document.createElement(tag);
+    element.classList.add(className);
+    return element;
```

Comment:
Let's add space before the return 🙌
```suggestion
    element.classList.add(className);
    
    return element;
```